scripts/create_testenv.sh

```diff
@@ -2,44 +2,41 @@
 
 set -ex # fail on first error, print commands
 
-while test $# -gt 0
-do
-    case "$1" in
-        --global)
-            GLOBAL=1
-            ;;
-        --no-setup)
-            NO_SETUP=1
-            ;;
-    esac
-    shift
+while test $# -gt 0; do
+  case "$1" in
+  --global)
+    GLOBAL=1
+    ;;
+  --no-setup)
+    NO_SETUP=1
+    ;;
+  esac
+  shift
 done
 
 command -v conda >/dev/null 2>&1 || {
-  echo "Requires conda but it is not installed.  Run install_miniconda.sh." >&2;
-  exit 1;
+  echo "Requires conda but it is not installed.  Run install_miniconda.sh." >&2
+  exit 1
 }
 
-ENVNAME="${ENVNAME:-testenv}" # if no ENVNAME is specified, use testenv
+ENVNAME="${ENVNAME:-testenv}"         # if no ENVNAME is specified, use testenv
 PYTHON_VERSION=${PYTHON_VERSION:-3.6} # if no python specified, use 3.6
 
-if [ -z ${GLOBAL} ]
-then
-    if conda env list | grep -q ${ENVNAME}
-    then
-      echo "Environment ${ENVNAME} already exists, keeping up to date"
-    else
-      conda create -n ${ENVNAME} --yes pip python=${PYTHON_VERSION}
-    fi
-    source activate ${ENVNAME}
+if [ -z ${GLOBAL} ]; then
+  if conda env list | grep -q ${ENVNAME}; then
+    echo "Environment ${ENVNAME} already exists, keeping up to date"
+  else
+    conda create -n ${ENVNAME} --yes pip python=${PYTHON_VERSION}
+  fi
+  source activate ${ENVNAME}
 fi
 pip install --upgrade pip
 
 conda install --yes mkl-service
 conda install --yes -c conda-forge python-graphviz
 
-
-#  Install editable using the setup.py
+# Uninstall then pin Matplotlib until https://github.com/matplotlib/matplotlib/issues/15410 is resolved
+pip uninstall matplotlib --yes
```

Comment:
How is that pulling in the right matplotlib version?

Reply:
It's doing that inside the conda venv, so I think it should be able to uninstall the right version.
What's weird though is that once 3.2.1 is uninstalled, 3.2.0 is still there, as if there two versions in the same venv (you can't see that in my latest commit; I came across it in an earlier attempt)